core/src/OpenMP/Kokkos_OpenMP_Instance.hpp

```diff
@@ -122,8 +122,18 @@ inline bool execute_in_serial(OpenMP const& space = OpenMP()) {
 #else
   bool is_nested = static_cast<bool>(omp_get_nested());
 #endif
-  return (space.impl_internal_space_instance()->get_level() < omp_get_level() &&
-          !(is_nested && (omp_get_level() == 1)));
+  bool max_parallel_level_exceeded =
+      (space.impl_internal_space_instance()->get_level() < omp_get_level() &&
+       !(is_nested && (omp_get_level() == 1)));
+
+#ifndef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  if (max_parallel_level_exceeded)
+    Kokkos::abort(
+        "Kokkos::OpenMP: Nested parallelism requires the maximum active levels "
+        "to be larger than 1!");
```

Comment:
Wouldn't this error message be confusing in the case that `omp_get_max_active_levels()>1` but we've exceeded that level in the internal instance?

Reply:
The level stored in the internal instance is the OMP level the execution space instance was created on. If that level is higher than the maximum level, we are clearly already attempting to use nested parallelism. Let me try to improve the message a bit.

Reply:
I'd like to propose deprecating creating (OpenMP) execution space instances inside a parallel OpenMP region as well which would make this check clearer as well.